### DISCUSSION TITLE
Permissions on IP for specific prefixes

### DISCUSSION THREAD

**Discussion by aturbin-35** [netbox-community/netbox#13560] [Help Wanted!]
Hello,
I didn't found thread on the topic I'm struggling on so I start one here.

I want to allow a group of user to add/change/delete for ip and ip range only if the corresponding prefix exist.
Is that possible or does it even make sense on Netbox ?

For the moment the big issue is that a user working on 10.1.1.0/24 can create an IP in this subnet but can also create an ip in 10.2.1.0/24 even if the prefix exist or not (or worst, even if the prefix exist and it's not supposed to be modified by the user or the groupe of users).

Thank you for your help.

Regards

**Comment by aturbin-35** [comment]
Sorry I forgot some informations :
netbox community in v3.5.4

**Comment by nomad-cyanide** [comment]
Not completely sure, but I have not seen anything to enforce such a limitation within Netbox. 
I would have a check running each day or as often as makes sense, that there is no IP's that does not belong to a prefix and if there are, either raise a flag or fix it automatically. Personally I would raise a flag, so the users can learn from their mistakes.

**Reply by aturbin-35** [reply]
Ok well thank you for your reply.
I thought it would be "easily" done by using a smart constraint.
We are actually using phpIpam and this behaviour cannot exist in it, that's why I was convinced of the feasability on Netbox.

**Comment by pv2b** [comment]
You should be able to write a [Custom Validator](https://docs.netbox.dev/en/stable/customization/custom-validation/) to do what you need.
